force-app/main/default/classes/DMLOperations.cls

```diff
@@ -180,15 +186,11 @@ public with sharing class DMLOperations {
 	 * @param contactsToUpsert List of Contacts to be upserted and linked to accounts.
 	 */
 	public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
-		// Loop through each Contact
-
-		// Extract the Account Name from Contact's LastName
-
-		// Get the account based on the name or create one if it doesn't exist
-
-		// Link the Contact to its Account using the Account Id
-
-		// Upsert the Contacts
+		for(Contact contactToUpsert : contactsToUpsert){            // Loop through each Contact// Extract the Account Name from Contact's LastName 
+		    Account acct = upsertAccount(contactToUpsert.LastName); // Get the account based on the name or create one if it doesn't exist  
```

Comment:
It was a good thought to call the method that already sort of did this, but it does SOQL and DML, so you dont want to call it within a loop. Can you refactor this to not do SOQL and DML within a loop?